tests/http-adapter.spec.js

```diff
@@ -33,166 +29,169 @@ describe('HTTP adapter', function () {
 
-      // Turn on `serializeRequests` and make some requests
+      const fetch1 = adapter.get(path1)
       adapter.serializeRequests = true
-      const fetch1 = adapter.get('/api/user/42/')
-      const fetch2 = adapter.get('/api/offer/9000/')
+      const fetch2 = adapter.get(path2)
 
-      // Tick microtask queue
+      // Tick microtask queue	
       await Promise.resolve().then(() => {})
 
-      expect(server.requests).toHaveLength(1)
-      expect(server.requests[0].url).toEqual('/api/user/42/')
-
-      const payload = {foo: 'bar', fiz: {biz: 'buz'}}
-      server.requests[0].respond(200, { "Content-Type": "application/json" }, JSON.stringify(payload))
+      expect(adapter._fetch.mock.calls.length).toBe(1)
+      expect(adapter._fetch.mock.calls[0][0].url.toString().includes(path1)).toBeTruthy()
 
       await fetch1
 
-      // Tick microtask queue
+      // Tick microtask queue	
       await Promise.resolve().then(() => {})
 
-      expect(server.requests).toHaveLength(2)
-      expect(server.requests[1].url).toEqual('/api/offer/9000/')
-
-      const payload2 = {foo2: 'bar', fiz2: {biz: 'buz'}}
-      server.requests[1].respond(200, { "Content-Type": "application/json" }, JSON.stringify(payload2))
+      expect(adapter._fetch.mock.calls.length).toBe(2)
+      expect(adapter._fetch.mock.calls[1][0].url.toString().includes(path2)).toBeTruthy()
 
       await fetch2
+      
+      done()
     })
 
-    it('parallel requests are restored after `serializeRequests` is toggled back', async function () {
-      server.autoRespond = false
-      server.respondImmediately = false
+    it('parallel requests are restored after `serializeRequests` is toggled back', async done => {
+      adapter._fetch = jest.fn().mockResolvedValue({})
 
-      // Turn on `serializeRequests` and make some requests
+      // Turn on `serializeRequests` and then make two parallel requests
       adapter.serializeRequests = true
-      const fetch1 = adapter.get('/api/user/42/')
-      const fetch2 = adapter.get('/api/offer/9000/')
+      const fetch1 = adapter.get(path1)
+      const fetch2 = adapter.get(path2)
 
-      // Respond to requests
+      // Tick microtask queue	
       await Promise.resolve().then(() => {})
-      const payload = {foo: 'bar', fiz: {biz: 'buz'}}
-      server.requests[0].respond(200, { "Content-Type": "application/json" }, JSON.stringify(payload))
+
+      expect(adapter._fetch.mock.calls.length).toBe(1)
+      expect(adapter._fetch.mock.calls[0][0].url.toString().includes(path1)).toBeTruthy()
+
       await fetch1
 
+      // Tick microtask queue	
       await Promise.resolve().then(() => {})
-      const payload2 = {foo2: 'bar', fiz2: {biz: 'buz'}}
-      server.requests[1].respond(200, { "Content-Type": "application/json" }, JSON.stringify(payload))
+
+      expect(adapter._fetch.mock.calls.length).toBe(2)
+      expect(adapter._fetch.mock.calls[1][0].url.toString().includes(path2)).toBeTruthy()
```

Comment:
Same as above. Use the API, Luke!

![Use the Force](https://media.giphy.com/media/l1KsOGE23suZQeCfm/giphy.gif)

Reply:
Perfect. I'll call the right api functions, remove the "done"s, format files, and rename some stuff!